docs/portingguide/src/howto/nogc.md

```diff
@@ -1,50 +1,277 @@
+
+However, in certain cases, we may need to have metadata globally for the entire heap space. The classic example is the valid-object bit metadata which tells us if an arbitrary address is allocated/managed by MMTk. Such a metadata, which spans multiple policies, is referred to as "global" metadata.
+
+For example, the *Forwarding bits and pointer* metadata is a local metadata used by copying policies to store forwarding bits (2-bits) and forwarding pointers (word size). Often runtimes require word-aligned addresses which means we can use the last two bits in the object header (due to alignment) and the entire object header to store the forwarding bits and pointer respectively. This metadata is almost always in the header.
+
+We recommend going through the [list of metadata specifications](https://www.mmtk.io/mmtk-core/public-doc/vm/trait.ObjectModel.html#required-associated-consts) that are defined by MMTk. You should set them to locations that are appropriate for your runtime.
+
+#### `ObjectReference` vs `Address`
+
+A key principle in MMTk is the distinction between [`ObjectReference`](https://www.mmtk.io/mmtk-core/public-doc/util/address/struct.ObjectReference.html) and [`Address`](https://www.mmtk.io/mmtk-core/public-doc/util/address/struct.Address.html). The idea is that very few operations are allowed on an `ObjectReference`. For example, MMTk does not allow address arithmetic on `ObjectReference`s. This allows us to preserve memory-safety, only performing unsafe operations when required, and gives us a cleaner and more flexible abstraction to work with as it can allow object handles or offsets etc. `Address`, on the other hand, represents an arbitrary machine address.
+
+You might be interested in reading the *Demystifying Magic: High-level Low-level Programming* paper[^3] which describes the above in more detail.
+
+[^3]: https://users.cecs.anu.edu.au/~steveb/pubs/papers/vmmagic-vee-2009.pdf
+
+#### Miscellaneous configuration options
+
+There are many constants in the `ObjectModel` trait that can be overridden in your binding in order to meet your runtime's requirements. For example, the `OBJECT_REF_OFFSET_LOWER_BOUND` constant which defines the minimum offset from allocation result start (i.e. the address that MMTk will return to the runtime) and the actual start of the object, i.e. the `ObjectReference`. In other words, the constant represents the minimum offset from the allocation result start such that the following invariant always holds:
+
+    OBJECT_REFERENCE >= ALLOCATION_RESULT_START + OFFSET
+
+We recommend going through the [list of constants in the documentation](https://www.mmtk.io/mmtk-core/public-doc/vm/trait.ObjectModel.html) and seeing if the default values suit your runtime's semantics, changing them if required.
+
+## MMTk initialization
+Now that we have most of the boilerplate set up, the next step is to initialize MMTk so that we can start allocating objects.
+
+### Runtime changes
```

Comment:
I feel it is a bit strange about the subsection headers 'Runtime changes' vs 'Rust changes'. A runtime can be implemented in Rust, and that will make the headers really confusing. I think you call them 'Runtime-side changes' and 'MMTk-side changes' if you refer to the changes in the two different directories in the binding.

Reply:
Yep that makes sense. I'll make the change.

Reply:
Also the new text makes the assumption that the runtime is implemented in native languages, such as C/C++. You may want to make it explicit that the guide uses C/C++ as an example for the runtime, but we do not require the runtime to be implemented in certain languages to work with MMTK.

Reply:
Okay I'll add that too.

Reply:
Done.